### PR TITLE
Support apns http2 node apn version

### PR DESCRIPTION
**Pull request by superkhau**
Helping land #133 

cc @iliraga @raymondfeng

### CODE REVIEW

**Comment by superkhau** [comment]
@raymondfeng The tests are failing because the new node-apns uses es6 enhanced object literal syntax: https://github.com/node-apn/node-apn/blob/master/index.js#L28. Merging this will mean we're not supporting node 10/12 anymore. Will ping you in the morning to confirm how to move forward.

**Comment by raymondfeng** [comment]
Hmm, that's annoying. I guess we can potentially release a new major version that requires Node 4 or later.

**Comment by superkhau** [comment]
I agree with your proposed solution because I don't see simpler way around this without forking and maintaining 2 versions. We're dropping 10 soon anyways and 12 in like another month or so: https://github.com/strongloop/loopback/issues/2807. The question is whether you want to just land this now or wait for LB to officially drop support.

@bajtos Thoughts?

**Comment by bajtos** [comment]
We should start a new major version of loopback-component-push first and drop support for Node v0.10/v0.12 there. This pull request should be landed on the new master and should not affect current version line 1.x.

Dropping support for Node v0.10/v0.12 is basically waiting for our CI build-framework to recognize `engines.node` field and don't run tests on older versions of Node (internal reference: https://github.com/strongloop-internal/build-framework/pull/12).

**Comment by bajtos** [comment]
Now that #135 was landed, we can move forward with this patch too. @superkhau

File: test/apns.provider.test.js [hunk 1]
```diff
@@ -24,7 +34,7 @@ describe('APNS provider', function() {
     afterEach(mockery.tearDown);
 
     it('sends Notification as an APN message', function(done) {
-      givenProviderWithConfig();
+      givenProviderWithConfig(defaultConfiguration);
```
Comment (bajtos) [inline test/apns.provider.test.js]: The idea of `givenProviderWithConfig` is to create a provider with sensible default configuration, so that we don't have to repeat ourselves in the tests and also don't have to change all tests when the default configuration changes like here.

Please change `givenProviderWithConfig` to set sensible `apns` defaults and revert changes in this file.

File: test/apns.provider.test.js [hunk 2]
```diff
@@ -71,23 +81,26 @@ describe('APNS provider', function() {
...
-      var devices = [aDeviceToken];
-      mockery.emitFeedback(devices);
+      provider.on('devicesGone', eventSpy);
+      provider.pushNotification(notification, aDeviceToken);
 
-      expect(eventSpy.args[0]).to.deep.equal([devices]);
-      done();
+      // HACK: Timeout does not work at this point
+      Promise.resolve(true).then(function() {
+        assert(eventSpy.called);
```
Comment (bajtos) [inline test/apns.provider.test.js]: You lost the assertion on notifications reported:

```js
expect(eventSpy.args[0]).to.deep.equal([devices]);
```

File: test/push-notification.test.js [hunk 3]
```diff
@@ -2,83 +2,87 @@
...
-            certData: objectMother.apnsDevCert(),
-            keyData: objectMother.apnsDevKey(),
-            pushOptions: {
-            },
-            feedbackOptions: {
-              batchFeedback: true,
-              interval: 300,
-            },
+    // Sign up an application
+    Application.register('test-user', 'TestApp', {
```
Comment (bajtos) [inline test/push-notification.test.js]: This looks like an unrelated whitespace change to me, can we please revert?

File: test/push-notification.test.js [hunk 4]
```diff
@@ -2,83 +2,87 @@
...
-            PushModel.dataSource.connector.applicationsCache.set(
-              application.id,
-              {
+          PushModel.dataSource.connector.applicationsCache.set(
+              application.id, {
                 memory: {
                   pushNotification: function(notification, deviceToken) {
-                    // console.log(notification, deviceToken);
-                    assert.equal(deviceToken, result.deviceToken);
+                    assert.equal(deviceToken, deviceToken);
```
Comment (bajtos) [inline test/push-notification.test.js]: Asserting that `X == X` makes no sense as it always passes. Please revert this line.

**Comment by superkhau** [comment]
@iliraga Would you like to address the comment above by @bajtos in your PR at #133? You can push your updates there and I can help you reland it again when you ping again.

**Comment by iliraga** [comment]
@superkhau I will address those issues and deliver a solution within this week if possible.

**Comment by superkhau** [comment]
@iliraga No problem, just ping again when you need review/help landing after your updates. Let's continue the conversation in your other PR. I'm going to close this and try to land your original PR since you'll be pushing updates anyways. Please rebase it with master while you're at it.
